refactor(test): cleanup of L1 test structure and documentation

aliersh opened the pull request:
This PR refactors a portion of the test suite within the `/L1` folder as part of a broader initiative to improve consistency, clarity, and structure across the codebase.

### Changes applied to each file:
- Centralized common setup into `*_TestInit` contracts
- Grouped tests by function into dedicated contracts inheriting from the init contract
- Standardized doc tags by adding `@title` and `@notice` headers
- Replaced `@dev` with `@notice` where applicable
- Applied 100-character line limit to all inline comments
- Maintained exact logic and comment wording per contribution guidelines

### Progress

Refactored 13 of 13 test files (**100% complete**)

- [x] `DataAvailabilityChallenge.t.sol`
- [x] `ETHLockbox.t.sol`
- [x] `L1CrossDomainMessenger.t.sol`
- [x] `L1ERC721Bridge.t.sol`
- [x] `L1StandardBridge.t.sol`
- [x] `OptimismPortal2.t.sol`
- [x] `OPContractsManager.t.sol`
- [x] `ProtocolVersions.t.sol`
- [x] `ProxyAdminOwnedBase.t.sol` ✅ No change needed
- [x] `ResourceMetering.t.sol`
- [x] `StandardValidator.t.sol` Refactored in #16068
- [x] `SuperchainConfig.t.sol`
- [x] `SystemConfig.t.sol`

aliersh committed:
refactor(test): improve L1CrossDomainMessenger test structure and documentation

- consolidate test initialization into a single L1CrossDomainMessenger_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- convert @dev to @notice tags where appropriate
- ensure comment lines stay within 100 character limit
- remove duplicated ForgeArtifacts and StorageSlot imports
- add virtual modifier to setup function

aliersh committed:
refactor(test): rename L1CrossDomainMessenger_Test to L1CrossDomainMessenger_Unclassified_Test

aliersh committed:
refactor(test): improve L1StandardBridge test structure and documentation

- consolidate test initialization into a single L1StandardBridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- group tests not directly tied to a base contract function or are testing multiple functions into L1StandardBridge_Unclassified_Test
- rename setUp to _setupPauseBridge as a helper function for paused state tests
- refactor test function names to follow naming convention
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve ResourceMetering test structure and documentation

- consolidate test initialization into a single ResourceMetering_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- refactor test function names to follow naming convention
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve OPContractManager test structure and documentation

- consolidate test initialization into a single OPContractManager_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- refactor test function names to follow naming convention
- add @title and @notice tags to all test contracts
- convert @dev tags to @notice where appropriate
- ensure comment lines stay within 100 character limit

aliersh committed:
fix(test): restore correct implementation of test_pause_callsSuperchainConfig_succeeds

smartcontracts commented on packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol:
```diff
@@ -113,14 +125,129 @@ contract L1CrossDomainMessenger_Test is CommonTest {
+        // Verify that the systemConfig was updated.
+        assertEq(address(l1CrossDomainMessenger.systemConfig()), address(newSystemConfig));
+
+        // Verify that the spacer was cleared.
```
These three lines can be removed, this assertion is no longer needed

aliersh replied:
Resolved in [f33a7f9](https://github.com/ethereum-optimism/optimism/pull/15903/commits/f33a7f918f7ac2c1a782867ebea0ca95f552475c)

aliersh committed:
fix(test): cleanup L1CrossDomainMessenger and restore OPContractManager test

- Remove obsolete systemConfig assertion in L1CrossDomainMessenger test
- Restore test_verifyOpcmCorrectness_succeeds in OPContractManager test suite

smartcontracts commented:
/ci authorize f33a7f918f7ac2c1a782867ebea0ca95f552475c

aliersh committed:
fix(test): fix failing upgrade tests in OPContractsManager and L1CrossDomainMessenger

- Remove skipIfNotOpFork conditions in OPContractsManager upgrade tests
- Add runUpgrade13UpgradeAndChecks calls in OPContractsManager upgrade tests
- Add systemConfig assertion in L1CrossDomainMessenger upgrade test

smartcontracts commented:
/ci authorize 9535b06421bd50acc0bcc936b0955563cf5a8470
